server/middleware/course-instances.coffee

```diff
@@ -232,3 +232,34 @@ module.exports =
       res.send(session.toObject({req}) for session in levelSessions)
     else
       res.send []
+
+  fetchPeerProjects: wrap (req, res) ->
+    courseInstance = yield database.getDocFromHandle(req, CourseInstance)
+    if not courseInstance
+      throw new errors.NotFound('Course Instance not found.')
+
+    unless courseInstance.get('ownerID').equals(req.user._id) or _.any(courseInstance.get('members'), (id) -> id.equals req.user._id)
+      throw new errors.Forbidden('You must be a member of a the given course instance')
+
+    classroom = yield Classroom.findById(courseInstance.get('classroomID'))
+    if not classroom
+      throw new errors.NotFound('Classroom not found.')
+
+    levelOriginalQueries = []
+    for course in classroom.get('courses') when course._id.equals(courseInstance.get('courseID'))
+      for level in course.levels when level.shareable is 'project'
+        levelOriginalQueries.push({
+          'level.original': level.original + '',
+          codeLanguage: level.primerLanguage or classroom.get('aceConfig.language') #TODO: Is this appropriate to include for projects?
```

Comment:
Oh probably not, I thought you meant the whole line. Can delete the bit about primer language, I think.